CMakeLists.txt

```diff
@@ -103,6 +103,11 @@ target_include_directories(AdePT
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 target_link_libraries(AdePT INTERFACE CopCore::CopCore)
 
+# Optional library to activate NVTX annotations for profiling:
+add_library(NVTX INTERFACE)
```

Comment:
Shouldn't you have an `if` around this to make it optional?

Reply:
I don't know. It's still unused if you merged this right now.

I hadn't decided where to put the option to enable/disable NVTX. I just add `NVTX` to the list of libraries for the example I'm interested in. Alternatively, one could enable it for the entire project using
```
cmake -DNVTX=On <sourceDir>
```
and then the `if` would be useful. It may boil down to where you want to enable the tracing:
- In the `CMakeLists.txt` of the respective example
- Globally when configuring
- Or somewhere else

What do you think?

Reply:
I like the idea of the interface library. I thought it was used unconditionally. Having a CMake switch to turn this stuff on would be nice indeed. A global switch + per-example support should be fine, as not all examples have the tags anyway.

Reply:
Here's the switch:
```bash
cmake -DNVTX=On .
```